nncf/common/pruning/utils.py

```diff
@@ -636,4 +638,43 @@ def identity_mask_propagation(node: NNCFNode, graph: NNCFGraph) -> None:
         # In case for disconnected NNCFGraph
         input_masks = [None]
     assert len(input_masks) == 1
-    node.data['output_mask'] = input_masks[0]
+    input_mask = input_masks[0]
+
+    if isinstance(input_mask, dict):
+        input_mask = find_input_mask_for_node(input_mask, node)
+    node.data['output_mask'] = input_mask
+
+
+def match_multiple_output_masks(output_masks: List[SymbolicMask], output_edges: List[NNCFGraphEdge],
+                                chunk_axis: int) -> Dict['str', SymbolicMask]:
+    """
+    Match multiple input mask to each next nodes.
+
+    :param output_masks: Given output masks.
+    :return: Matched output mask for each next node.
+    """
+    output_shapes = [edge.tensor_shape[chunk_axis] for edge in output_edges]
+    next_nodes = [edge.to_node for edge in output_edges]
+    result_masks = {node.node_name: None for node in next_nodes}
+
+    if len(set(output_shapes)) == 1:
+        for i, split_mask in enumerate(output_masks):
+            result_masks[next_nodes[i].node_name] = split_mask
+    else:
```

Comment:
What if output_shapes will be like [1, 1, 2]?

Reply:
I could not catch up yours. But in this case, I tested the case of like below.
Input of chunk operation : C=16 -> Output :C=6, C=6, C=4.
For output_shapes, I only consider the size of axis which the operation performed.
`output_shapes = [edge.tensor_shape[chunk_axis] for edge in output_edges]`
That's why I've made it possible to consider cases where there are multiple outputs of the same size.
Please tell me if you mean something other than this case.

Reply:
Ok, got it.


Can we do it without if statement in such manner?
```
result_mask = dict()
tmp_out_masks = out_masks.copy()
for edge in output_edges:
    node_name = edge.to_node.node_name
    idx = tmp_out_masks.index(edge.tesnsor_shape[chunk_axis])
    result_mask[node_name] = tmp_out_masks.pop(idx)

return result_mask
```

Reply:
I fixed it following your advice. Please let me know if there is anything else to fix.